src/xma/src/xmaapi/xmaxclbin.cpp

```diff
@@ -375,19 +383,30 @@ int xma_xclbin_info_get(const char *buffer, XmaXclbinInfo *info)
     return XMA_SUCCESS;
 }
 
-int xma_xclbin_map2ddr(uint64_t bit_map, int32_t* ddr_bank)
+int xma_xclbin_map2ddr(uint64_t bit_map, int32_t* ddr_bank, bool has_mem_grps)
 {
     //64 bits based on MAX_DDR_MAP = 64
     int ddr_bank_idx = 0;
-    while (bit_map != 0)
-    {
-        if (bit_map & 1)
-        {
-            *ddr_bank = ddr_bank_idx;
-            return XMA_SUCCESS;
+    if (!has_mem_grps) {
+        while (bit_map != 0) {
+            if (bit_map & 1) {
+                *ddr_bank = ddr_bank_idx;
+                return XMA_SUCCESS;
+            }
+            ddr_bank_idx++;
+            bit_map = bit_map >> 1;
+        }
+    } else {//For Memory Groups use last group as default memory group; For HBM groups
```

Comment:
Oh, a comment!!!
The readers will be so happy!

Reply:
Understood. Will be better. This is old code and doing bug fix in this. Refactoring into proper cpp code is in progress. Will add you in those for review.